rust-examples/enum.rs

```diff
@@ -0,0 +1,63 @@
+#![feature(fundamental, lang_items, no_core)]
+#![allow(dead_code)]
+#![no_core]
+
+enum Tag {
+    A(i32),
+    B(i32),
+    C,
+}
+
+enum CTag {
+    A = -1,
+    B = 0,
+    C = 1,
+    D = 12,
+}
+
+fn main() {
+    let t = Tag::B(17);
+    let i = match t {
+        Tag::A(_) => 1,
+        Tag::B(i) => i,
+        _ => 3
+    };
+    wasm::print_i32(i); // (i32.const 17)
+
+    let t = CTag::A;
+    let i = match t {
+        i @ _ => i as i32
+    };
+    wasm::print_i32(i as i32); // (i32.const -1)
+    wasm::print_i32(CTag::D as i32); // (i32.const 12)
+}
+
+pub mod marker {
+    use clone::Clone;
+
+    #[lang = "sized"]
+    #[fundamental]
+    pub trait Sized { }
+
+    #[lang = "copy"]
+    pub trait Copy : Clone { }
+}
+
+pub mod clone {
```

Comment:
No they don't work yet. We're gonna have to do something about it, and think about how it should be exposed to wasm. Eg one rust module == one wasm module, or something else. A problem being wasm modules are not rust modules, and rust pub isn't exactly a wasm export, both are also designed to interop with other modules, or be called from javascript. That comes with additional constraints on which parameter types can be supported, etc. 
Still lots of questions left to answer before dealing with that I think (maybe after I get back from my august vacation), even though it is definitely an important topic.